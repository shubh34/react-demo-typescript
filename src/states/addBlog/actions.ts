
export const INIT_ADD_BLOG: string = "INIT_ADD_BLOG";
export const UPDATE_FORM_FIELD: string = "UPDATE_FORM_FIELD";
export const ADD_BLOG_CATEGORY: string = "ADD_BLOG_CATEGORY";
export const DELETE_BLOG_CATEGORY: string = "DELETE_BLOG_CATEGORY";

export const initAddBlogForm = () => {
    return {
        type: INIT_ADD_BLOG,
    }
}

export const updateFormField = (name: string, value: string) => {
    return {
        type: UPDATE_FORM_FIELD,
        field: {
            name,
            value,
        }
    }
}

export const addCategory = (category: string) => {
    return {
        type: ADD_BLOG_CATEGORY,
        category,
    }
}

export const deleteCategory = (category: string) => {
    return {
        type: DELETE_BLOG_CATEGORY,
        category,
    }
}


