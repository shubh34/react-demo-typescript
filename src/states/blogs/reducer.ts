import { IBlogsState } from "./types";
import { ADD_BLOG } from "./action";

const init: IBlogsState = {
    blogs: []
};


const reducer = (state = init, action: any) : IBlogsState => {
	switch (action.type) {
	case ADD_BLOG: 
		return {
			...state,
			blogs: [...state.blogs, action.blog],
		}
	default:
		console.log('state', state);
		return state;
	}
};
export default reducer;
