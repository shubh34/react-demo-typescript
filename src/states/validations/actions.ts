
export const INIT_VALIDATION: string = "INIT_VALIDATION";
export const UPDATE_VALIDATION: string = "UPDATE_VALIDATION";
export const RESET_VALIDATION: string = "RESET_VALIDATION";

export const initValidation = (name: string, isValid: boolean) => {
    return {
        type: INIT_VALIDATION,
        field: {
            name,
            isValid,
        }
    }
}

export const updateValidation = (name: string, isValid: boolean) => {
    return {
        type: UPDATE_VALIDATION,
        field: {
            name,
            isValid,
        }
    }
}
export const resetValidation = () => {
    return {
        type: RESET_VALIDATION,
    }
}


