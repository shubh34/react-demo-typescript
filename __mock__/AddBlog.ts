import { IAddBlogState } from "../src/states/addBlog/types";

export const addBlogDefaultState: IAddBlogState = {
    content: '',
    title: '',
    categories: ['test'],
 }

 