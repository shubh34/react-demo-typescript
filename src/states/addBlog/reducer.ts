import { IAddBlogState } from './types';
import { INIT_ADD_BLOG, UPDATE_FORM_FIELD, ADD_BLOG_CATEGORY, DELETE_BLOG_CATEGORY } from './actions';


const addBlogInitialState: IAddBlogState = {
   title:  '',
   categories: [],
   content: '',
}



const reducer = (state = addBlogInitialState, action: any): IAddBlogState => {
	switch (action.type) {
    case INIT_ADD_BLOG : 
        return {
            ...state,
            ...addBlogInitialState,
        }
    case UPDATE_FORM_FIELD: 
        return {
            ...state,
            [action.field.name]: action.field.value,
        }
    case ADD_BLOG_CATEGORY: 
        return {
            ...state,
            categories: [...state.categories, action.category]
        }
    case DELETE_BLOG_CATEGORY: 
        return {
            ...state,
            categories: state.categories.filter(category => category !== action.category)
        }
	default:

		return state;
	}
};
export default reducer;
