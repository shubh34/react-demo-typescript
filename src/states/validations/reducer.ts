import { IValidationState } from './types';
import { RESET_VALIDATION, INIT_VALIDATION, UPDATE_VALIDATION } from './actions';

const initialValidation: IValidationState = {
    form: {
        title: {
            isValid :  false, 
        },
        categories: {
            isValid :  false, 
        },
        content: {
            isValid: false,
        }
    },
}



const reducer = (state = initialValidation, action: any): IValidationState => {
	switch (action.type) {
    case RESET_VALIDATION : 
        return {
            ...state,
            ...initialValidation,
        }
    case INIT_VALIDATION: 
        return {
            ...state,
            form: {
                ...state.form,
                [action.field.name]: {
                    isValid: action.field.isValid,
                }
            }
        }
    case UPDATE_VALIDATION: 
        return {
            ...state,
            form: {
                ...state.form,
                [action.field.name]: {
                    isValid: action.field.isValid,
                }
            }
        }
     
	default:
		return state;
	}
};
export default reducer;
