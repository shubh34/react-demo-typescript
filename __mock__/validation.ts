import { IValidationState } from "../src/states/validations/types";

export const initiaValidation: IValidationState = {
    form: {
        title: {
            isValid :  false, 
        },
        categories: {
            isValid :  false, 
        },
        content: {
            isValid: false,
        }
    },
}
export const formValid: IValidationState= {
    form: {
        title: {
            isValid :  true, 
        },
        categories: {
            isValid :  true, 
        },
        content: {
            isValid: true,
        }
    },
}