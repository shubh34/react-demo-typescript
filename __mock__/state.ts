import {rawBlogLists} from './blogs';
import { initiaValidation, formValid } from './validation';
import { IRootState } from '../src/store';

export const defaultState: IRootState = {
    blogListing: {
        blogs: rawBlogLists
     },
     validation: initiaValidation,
     addBlog: {
        content: '',
        title: '',
        categories: ['test'],
     }
};


export const formValidState: IRootState = {
    ...defaultState,
    validation: formValid,
};