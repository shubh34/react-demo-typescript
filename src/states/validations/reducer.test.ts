import reducer from  './reducer';
import { formValid } from '../../../__mock__/validation';
import { resetValidation, updateValidation, initValidation } from './actions';

describe('Reducer validation', () => {
    it('should reset validation', () => {
       expect(reducer(formValid, resetValidation())).toMatchSnapshot();
    });
    it('should update fieldValidation validation', () => {
        expect(reducer(formValid, updateValidation('title', false))).toMatchSnapshot();
     });
     it('should update fieldValidation validation', () => {
        expect(reducer(formValid, initValidation('title', true))).toMatchSnapshot();
     });
})