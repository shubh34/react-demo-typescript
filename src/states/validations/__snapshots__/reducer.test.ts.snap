// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Reducer validation should reset validation 1`] = `
Object {
  "form": Object {
    "categories": Object {
      "isValid": false,
    },
    "content": Object {
      "isValid": false,
    },
    "title": Object {
      "isValid": false,
    },
  },
}
`;

exports[`Reducer validation should update fieldValidation validation 1`] = `
Object {
  "form": Object {
    "categories": Object {
      "isValid": true,
    },
    "content": Object {
      "isValid": true,
    },
    "title": Object {
      "isValid": false,
    },
  },
}
`;

exports[`Reducer validation should update fieldValidation validation 2`] = `
Object {
  "form": Object {
    "categories": Object {
      "isValid": true,
    },
    "content": Object {
      "isValid": true,
    },
    "title": Object {
      "isValid": true,
    },
  },
}
`;
